bot/cogs/filtering.py

```diff
@@ -126,6 +132,41 @@ async def on_message_edit(self, before: Message, after: Message) -> None:
             delta = relativedelta(after.edited_at, before.edited_at).microseconds
         await self._filter_message(after, delta)
 
+    @Cog.listener('on_message')
+    async def bad_words_in_name(self, msg: Message) -> None:
+        """Check bad words from user display name. When there is more than 3 days after last alert, send new alert."""
+        if await self.name_alerts.contains(msg.author.id):
+            last_alert = parser.isoparse(await self.name_alerts.get(msg.author.id))
```

Comment:
You can save yourself a call to Redis if you just `get` the value and abort if it gives `None`.

Keep in mind that this event will fire for *every* message sent in the guild, so if we can avoid the superfluous Redis call, we probably should. In fact, it would likely make more sense to poll Redis if and only if a `match` is found `(L153)`.

In other words, instead of:

1. See if an alert can be sent for given user (regardless of nickname)
2. If so, analyse author name

We can do:

1. Analyse author name for issues
2. If issues exist, check if alert should be sent

I'm not sure exactly how performant the cache is and whether superfluous calls to it are any problem at all, but given considering that this event fires *very* often, my instinct is to at least consider the potential performance implications it may have.